app/repository/database.go

```diff
@@ -88,24 +88,24 @@ func (c *dbConnection) ListActivityPrograms(userID string) ([]*model.ActivityPro
 	return aps, nil
 }
 
-func (c *dbConnection) FindRoute(id model.RouteID) (*model.Route, error) {
-	r := &model.Route{
-		ID: id,
-	}
-	err := c.db.Find(r).Error
-	if err != nil {
+func (c *dbConnection) FindRoute(userID string, seqNo model.RouteSeqNo) (*model.Route, error) {
+	r := &model.Route{}
+	err := c.db.Where("user_id = ? and seq_no = ?", userID, seqNo).Find(&r).Error
+	if err != nil && !errors.Is(err, gorm.ErrRecordNotFound) {
 		return nil, err
+	} else if errors.Is(err, gorm.ErrRecordNotFound) {
+		return nil, nil
 	}
 	return r, nil
 }
 
-func (c *dbConnection) FindMaxRouteID() (model.RouteID, error) {
+func (c *dbConnection) FindRouteMaxSeqNo(userID string) (model.RouteSeqNo, error) {
 	r := &model.Route{}
-	err := c.db.Last(r).Error
+	err := c.db.Limit(1).Order("seq_no DESC").Where("user_id = ?", userID).Find(r).Error
```

Comment:
```suggestion
	err := c.db.Order("seq_no DESC").Last().Where("user_id = ?", userID).Find(r).Error
```

@oshikun 
ここのOrder Byありがとうございます。ナイスです。
Limit(1)とLast()は同じ動きをするので、Lastで大丈夫です。